### DISCUSSION TITLE
Paste and clean up text

### DISCUSSION THREAD

**Discussion by its-me-steve** [p0deje/Maccy#884] [Ideas]
First of all, love the app. It's a workhorse for me, use it every day, all the time at work.

I migrated to Mac a while ago and back on Windows I used Ditto as a clipboard manager. And it had some features I used a lot. It had the ability to do with some text cleanup when pasting, such as removing line feeds, or trimming white space (double spaces etc). If something like this could be integrated with Maccy it'd make it even more perfect.

**Comment by p0deje** [comment]
This is primarily solved by Shortcuts - you can create a custom workflow that will take the text from the selected item in Maccy, clean it up as you want, and paste.

For example:
<img width="1312" alt="Screenshot 2024-09-19 at 07 07 05" src="https://github.com/user-attachments/assets/9a595ba4-48cf-451b-a459-00a07dfdb145">

**Reply by its-me-steve** [reply]
Thanks, after some fiddling with permissions I got it working! Excellent workaround. I use Shortcuts for a different other stuff and it's really powerful.
